File: AngularBooking/ClientApp/src/app/core/_models/entity/booking.ts
import { BookingStatus } from "./booking-status";
import { Showing } from "./showing";
import { Customer } from "./customer";
import { BookingItem } from "./booking-item";

export class Booking {
  constructor(
    public id?: number,
    public bookedDate?: string,
    public status?: BookingStatus,
    public showingId?: number,
    public customerId?: number,
    public bookingItems?: BookingItem[]

  ) { }

  public totalPrice(): number {
    if (this.bookingItems) {
      var total = 0;
      for (let bookingItem of this.bookingItems) {
        total += bookingItem.agreedPrice;
      }
      return total;
    }
    return 0;
  }

}

File: AngularBooking/Models/Booking.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace AngularBooking.Models
{
    public class Booking : IModel
    {
        public int Id { get; set; }
        public int ShowingId { get; set; }
        public int CustomerId { get; set; }
        public DateTime BookedDate { get; set; }
        public BookingStatus Status { get; set; }
        public string AdditionalDetails { get; set; }

        // navigation
        public Showing Showing { get; set; }
        public Customer Customer { get; set; }
        public List<BookingItem> BookingItems { get; set; }
    }
}
